### PR TITLE
fixed braziers for wintertodt + rs2grounditem.loot

### PR DESCRIPTION
**Pull request by ObserveObserver**
fixed braziers for wintertodt, fixed bug in wintertodt where player would be stuck in WAITING state (i think i did, anyway), fixed rs2grounditem.loot for 50 range instead of 255. tutorialisland fix still.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java [hunk 1]
```diff
@@ -363,14 +377,14 @@ private boolean handleBankLogic(MWintertodtConfig config) {
...
         Rs2Bank.depositAll();
         int foodCount = (int) Rs2Inventory.getInventoryFood().stream().count();
-//        if (config.fixBrazier()) {
-//            Rs2Bank.withdrawX(true, "hammer", 1);
-//        }
+        if (config.fixBrazier()) {
+            Rs2Bank.withdrawX(true, "hammer", 1);
+        }
         if (!Rs2Equipment.hasEquipped(ItemID.BRUMA_TORCH)) {
-            Rs2Bank.withdrawX(true, "tinderbox", 1, true);
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java]: why did you remove the true statement here?

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java [hunk 2]
```diff
@@ -354,7 +368,7 @@ private boolean handleBankLogic(MWintertodtConfig config) {
         WorldPoint bankLocation = new WorldPoint(1640, 3944, 0);
         if (Rs2Player.getWorldLocation().distanceTo(bankLocation) > 6) {
             Rs2Walker.walkTo(bankLocation);
-            Rs2Player.waitForWalking();
+            sleepUntil(() -> Rs2GameObject.findChest() != null);
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java]: this is more resource intensive, not sure if this is really needed.

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java [hunk 3]
```diff
@@ -95,26 +94,28 @@ public boolean run(MWintertodtConfig config) {
 
                 if (!needBanking) {
                     if (!isWintertodtAlive) {
-                        if (state != State.ENTER_ROOM && state != State.WAITING && state != State.BANKING) {
+                        if ((state != State.ENTER_ROOM && state != State.WAITING && state != State.BANKING)
+                        || Rs2Inventory.hasItem("supply crate")) {
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java]: use SUPPLY_CRATE to be consistent

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java [hunk 4]
```diff
@@ -381,6 +395,7 @@ private boolean handleBankLogic(MWintertodtConfig config) {
             return true;
         }
         Rs2Bank.withdrawX(config.food().getId(), config.foodAmount() - foodCount);
-        return sleepUntilTrue(() -> Rs2Inventory.hasItemAmount(config.food().getName(), config.foodAmount(), false, true), 100, 5000);
+        sleepUntil(() -> Rs2Inventory.hasItemAmount(config.food().getName(), config.foodAmount(), false, true));
+        return false;
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java]: this is weird, why are you always returning false?

Comment (ObserveObserver) [reply runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java]: i have no idea where that came from, maybe some check i deleted early on

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java (hunk 2 above)
Comment (ObserveObserver) [reply runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java]: it was an attempt to get to open crates after it saw the bank such that it loads into loot tracker plugin (as it only loads into it after you're out of wintertodt doors)

**Comment by chsami** [comment]
I will close this one out as it no longer contains any relevant code to the original PR. I am keen to integrate your tutorial island changes to the script, if you want to PR those seperately
